octodns/provider/easydns.py

```diff
@@ -0,0 +1,453 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from collections import defaultdict
+from requests import Session
+from time import sleep
+import logging
+import base64
+
+from ..record import Record
+from .base import BaseProvider
+
+
+class EasyDNSClientException(Exception):
+    pass
+
+
+class EasyDNSClientBadRequest(EasyDNSClientException):
+
+    def __init__(self):
+        super(EasyDNSClientBadRequest, self).__init__('Bad request')
+
+
+class EasyDNSClientNotFound(EasyDNSClientException):
+
+    def __init__(self):
+        super(EasyDNSClientNotFound, self).__init__('Not Found')
+
+
+class EasyDNSClientUnauthorized(EasyDNSClientException):
+
+    def __init__(self):
+        super(EasyDNSClientUnauthorized, self).__init__('Unauthorized')
+
+
+class EasyDNSClient(object):
+    # EasyDNS Sandbox API
+    SANDBOX = 'https://sandbox.rest.easydns.net'
+    # EasyDNS Live API
+    LIVE = 'https://rest.easydns.net'
+    # Default Currency CAD
+    defaultCurrency = 'CAD'
+    # Domain Portfolio
+    domainPortfolio = 'myport'
+
+    def __init__(self, token, apikey, currency, portfolio, sandbox):
+        self.log = logging.getLogger('EasyDNSProvider[{}]'.format(id))
+        self.token = token
+        self.apikey = apikey
+        self.defaultCurrency = currency
```

Comment:
octoDNS uses snake_case, rather than camelCase. Please make a pass through and convert things for consistency with the existing codebase.

Reply:
I appreciate the feedback.  I've removed the camelCase (bad habit's sometimes sneak in) and will commit an update shortly.